source/common/upstream/upstream_impl.cc

```diff
@@ -1752,9 +1774,9 @@ void ClusterImplBase::reloadHealthyHostsHelper(const HostSharedPtr&) {
     HostVectorConstSharedPtr hosts_copy = std::make_shared<HostVector>(host_set->hosts());
 
     HostsPerLocalityConstSharedPtr hosts_per_locality_copy = host_set->hostsPerLocality().clone();
-    prioritySet().updateHosts(priority,
-                              HostSetImpl::partitionHosts(hosts_copy, hosts_per_locality_copy),
-                              host_set->localityWeights(), {}, {}, absl::nullopt, absl::nullopt);
+    prioritySet().updateHosts(
+        priority, HostSetImpl::partitionHosts(hosts_copy, hosts_per_locality_copy),
+        host_set->localityWeights(), {}, {}, random_.random(), absl::nullopt, absl::nullopt);
```

Comment:
Just to confirm, we're not thinking of any situations in which the seed should be set by the control plane? For WRR this is fine to randomize but consistent hashes might want this. Ignore this comment if this is only applicable to WRR style and similar ilk.

Reply:
Nope, the seed is not coming from the control-plane (the config doesn't have a knob for setting this).

IIUC consistent hashes (maglev and ring-hash) are not using the EDF-scheduler, but they are thread-aware LBs that normalize the locality+host weight into a [single normalized weight](https://github.com/envoyproxy/envoy/blob/20c7368afa9d686a109f9601ae1b9b6028b74b0a/source/common/upstream/thread_aware_lb_impl.cc#L85), and populate the data-structures using the normalized-weights.
From a first pass it seems that this code might be invoked also when consistent-hashes are used (and not sure why there's a need for an EDF-based locality scheduler in this case), so let me take a deeper look and I'll get back to you with an answer.

/wait-any

Reply:
Ok, I've validated that this code indeed is called when a maglev/ring-hash LB policy is used, so the EDF-scheduler is created, but it is not being used.
Seems wasteful IMHO, and could probably be optimized to not be done/allocated at all. Created #32145 to track this.

Reply:
OK, that doees seem weird, good catch.